src/js/imagery/imageryOptions.js

```diff
@@ -63,7 +63,11 @@ export const imageryOptions = [
     {
         type: "xyz",
         label: "XYZ Imagery",
-        params: [{key: "url", display: "XYZ URL"}],
+        params: [{
+            key: "url",
+            display: "XYZ URL",
+            validator: value => !(/https?:\/\/.*{-?[xyz]}\/{-?[xyz]}\/{-?[xyz]}.*/gi).test(value) ? "Invalid URL or URL doesn't contain {z}, {x} or {y}." : "",
```

Comment:
Dont require a slash between xyz.  x={x}&y={y} is also valid for example.

How does this look? `https?:\/\/.*{-?[xyz]}.*{-?[xyz]}.*{-?[xyz]?}.*`

Be more specific with you message.  You are checking for http, x, y, and z.  A generic "invalid url" does not help the user as much.

Reply:
https?:\/\/.*(?=.*{-?x})(?=.*{-?z})(?=.*{-?y}).*
this way ww can check if there is at least one {-?x}, one {-?y} and one {-?z} doesn't matter their position.